packages/plugin-connector/src/data-sources/sub-source-vars.js

```diff
@@ -0,0 +1,39 @@
+/**
+ * Replaces all ${var} patterns in a string with the matching EVIDENCE_VAR__[var] environment variables
+ * @param {string} queryString
+ * @returns {string}
+ */
+export const subSourceVariables = (queryString) => {
+	const envPrefix = 'EVIDENCE_VAR__';
+
+	if (queryString.length > 1024 * 1024) {
+		if (process.env.VITE_EVIDENCE_DEBUG) {
+			console.log(
+				`Skipping variable interpolation for file; has more than ${1024 * 1024} characters`
+			);
+		}
+		return queryString;
+	}
+
+	const validVars = Object.fromEntries(
+		Object.entries(process.env)
+			.filter(([k]) => k.startsWith(envPrefix))
+			.map(([k, v]) => {
+				const name = k.substring(envPrefix.length);
+				const value = v?.toString();
+				return [name, value];
+			})
+	);
+	let output = queryString;
+	const regex = RegExp(/\$\{(.+?)\}/, 'g');
+	let match;
+	while ((match = regex.exec(queryString)) !== null) {
+		if (match[1] in validVars) output = output.replace(match[0], validVars[match[1]]);
```

Comment:
This appears to throw
> │ > tsc --noEmit -p tsconfig.json && parcel build
│ src/data-sources/sub-source-vars.js(33,64): error TS2769: No overload matches this call.
│   The last overload gave the following error.
│     Argument of type 'string | undefined' is not assignable to parameter of type '(substring: s…
│       Type 'undefined' is not assignable to type '(substring: string, ...args: any[]) => string…